jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/QueryingLogGroups.kt

```diff
@@ -20,7 +20,22 @@ class QueryingLogGroups(private val project: Project) : CoroutineScope by Applic
             .startTime(queryStartEndDate.startDate.epochSecond)
             .build()
         val response = client.startQuery(request)
-        val queryId = response.queryId()
-        // TODO: Get the results of the query with qid
+        var queryId: String = response.queryId()
+        val fieldList = getFields(query)
+        QueryResultsWindow.getInstance(project).showResults(queryId, fieldList)
+    }
+
+    fun getFields(query: String): List<String> {
+        var fieldList: List<String>
+        if (query.contains("fields", ignoreCase = true)) {
+            var fields = query.substring(query.indexOf("fields", ignoreCase = true) + 7)
```

Comment:
Nit: indexOf and contains can be combined.
```
var fieldsIndex = query.indexOf("fields", ignoreCase = true)
if (fieldsIndex > 0) {
    var fields = query.substring(fieldsIndex + 7)
...
```

Also what happens if `fields` is a part of a filter string?